### PR TITLE
Ported to sbt 1.0

### PR DESCRIPTION
**Pull request by metasim**

### CODE REVIEW

**Comment by metasim** [comment]
@brkyvz Are you the owner of this project? 

If so, can you provide your thoughts on this PR, and how we might get a 1.0-compatible release published?

If not, who should we reach out to?

cc: @sullivan @thomasdziedzic

**Comment by metasim** [comment]
A binary version of this PR is available with this fragment:

```scala
resolvers += Resolver.bintrayIvyRepo("s22s", "sbt-plugins")
addSbtPlugin("org.spark-packages" % "sbt-spark-package" % "0.2.7-astraea.1")
```

or: https://bintray.com/s22s/sbt-plugins/sbt-spark-package

**Comment by sullis** [comment]
what is the status of this PR?

**Comment by metasim** [comment]
FYI: @sullis I sent an email in January to what I thought was the address for @brkyvz (guessed from profile), but never received a reply. I'm worried this has become abandonware, and maybe a community fork is needed. I hope I'm wrong, but I can't think of any other options at this point to keep it moving forward. Ideas?

**Comment by brkyvz** [comment]
@metasim @sullis Sorry folks, been very busy with the Spark 2.3 release. I'll take a look at this (test it etc) as soon as possible. Really appreciate the PR!

**Comment by mhamilton723** [comment]
+1 on this PR

**Comment by tejasmanohar** [comment]
@brkyvz Any update here?

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,8 @@
 language: scala
 scala:
-  - 2.10.4
+  - 2.12.4
```
Comment (zouzias) [inline .travis.yml:3]: You may want to upgrade to 2.12.7 here, see https://www.scala-lang.org/news/2.12.7

File: build.sbt [hunk 2]
```diff
@@ -1,20 +1,28 @@
-addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.3")
+lazy val root = (project in file("."))
+  .settings(
+    sbtPlugin := true,
+    scalaVersion := "2.12.4",
```
Comment (zouzias) [inline build.sbt:4]: Same here, 2.12.7

File: project/build.properties [hunk 3]
```diff
@@ -1 +1,2 @@
-sbt.version=0.13.6
\ No newline at end of file
+sbt.version=1.0.3
```
Comment (zouzias) [inline project/build.properties:1]: Maybe update to 1.2.3, see https://www.scala-sbt.org/download.html

File: src/main/scala/sbtsparkpackage/SparkPackagePlugin.scala [hunk 4]
```diff
@@ -57,7 +61,7 @@ object SparkPackagePlugin extends AutoPlugin {
       "as \"provided\" or not.")
 
     val defaultSPSettings = Seq(
-      sparkVersion := "1.4.0",
+      sparkVersion := "2.2.0",
```
Comment (zouzias) [inline src/main/scala/sbtsparkpackage/SparkPackagePlugin.scala:64]: Could be spark 2.3.2 as well.

File: build.sbt [hunk 5]
```diff
@@ -1,20 +1,28 @@
...
+//      ("org.scala-sbt" % "compiler-interface" % sv % "component").sources
+//    },
+    scriptedLaunchOpts := {
+      scriptedLaunchOpts.value ++
+        Seq("-Xmx1024M", "-Dplugin.version=" + version.value)
+    },
+    scriptedBufferLog := false
+  )
+
+addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.5")
```
Comment (zouzias) [inline build.sbt:26]: Update to 0.14.7

File: build.sbt [hunk 6]
```diff
@@ -1,20 +1,28 @@
-addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.3")
+lazy val root = (project in file("."))
+  .settings(
```
Comment (zouzias) [inline build.sbt:2]: If you update to sbt `1.2.3`, you need to add here `  .enablePlugins(SbtPlugin)`, see https://www.scala-sbt.org/1.0/docs/Testing-sbt-plugins.html#step+2%3A+SbtPlugin

File: build.sbt [hunk 7]
```diff
@@ -1,20 +1,28 @@
...
+    name := "sbt-spark-package",
+    organization := "org.spark-packages",
+    description := "sbt plugin to develop, use, and publish Spark Packages",
+    licenses := Seq(
+      "Apache-2.0" -> url("http://opensource.org/licenses/Apache-2.0")),
+    libraryDependencies ++= Seq(
+      "org.scalaj" %% "scalaj-http" % "2.3.0",
+      "org.apache.directory.studio" % "org.apache.commons.codec" % "1.8"
+    ),
+//    scalaCompilerBridgeSource := {
```
Comment (zouzias) [inline build.sbt:15]: You may remove the comments here.

File: src/sbt-test/sbt-spark-packages/R-packaging/build.sbt [hunk 8]
```diff
@@ -8,25 +8,25 @@ name := "r-packaging"
...
+    jarContentChecks(crossTarget.value / "r-packaging-assembly-0.1.jar", true)
   }
 }
 
-TaskKey[Unit]("checkBinJar") <<= (crossTarget) map { (crossTarget) =>
+TaskKey[Unit]("checkBinJar") := {
   IO.withTemporaryDirectory { dir =>
-    jarContentChecks(crossTarget / "r-packaging_2.10-0.1.jar", false)
-    validatePom(crossTarget / "r-packaging_2.10-0.1.pom", "awesome.test", "r-packaging_2.10")
+    jarContentChecks(crossTarget.value / "r-packaging_2.10-0.1.jar", false)
```
Comment (zouzias) [inline src/sbt-test/sbt-spark-packages/R-packaging/build.sbt:28]: Not sure if the versions here are up-to-date.

**Review by zouzias — COMMENTED** [review]
There are some version updates that could be done, but the PR looks good to me.

**Comment by metasim** [comment]
Closed due to lack of maintainer interest.

**Comment by rjurney** [comment]
@metasim is this something that could be taken up for Scala 2.12 and modern Spark? The GraphFrames project needs it to work with late model sbt.

**Pull request closed without merging**